File: spec/lua/test_bytes_pad_term.lua
local luaunit = require("luaunit")

require("bytes_pad_term")

TestBytesPadTerm = {}

function TestBytesPadTerm:test_bytes_pad_term()
    local r = BytesPadTerm:from_file("src/str_pad_term.bin")

    luaunit.assertEquals(r.str_pad, "str1")
    luaunit.assertEquals(r.str_term, "str2foo")
    luaunit.assertEquals(r.str_term_and_pad, "str+++3bar+++")
    luaunit.assertEquals(r.str_term_include, "str4baz@")
end

File: spec/python/test_bytes_pad_term.py
import unittest

from bytes_pad_term import BytesPadTerm

class TestBytesPadTerm(unittest.TestCase):
    def test_bytes_pad_term(self):
        r = BytesPadTerm.from_file("src/str_pad_term.bin")

        self.assertEqual(r.str_pad, b"str1")
        self.assertEqual(r.str_term, b"str2foo")
        self.assertEqual(r.str_term_and_pad, b"str+++3bar+++")
        self.assertEqual(r.str_term_include, b"str4baz@")
